### PR TITLE
cloudfront_distribution: Add support for cache_policy_id and origin_request_policy_id for behaviors

### LINKED ISSUE #290: cloudfront_distribution - Add support for setting cache and origin request policy ids in cache behavior
##### SUMMARY
Add support for setting cache and origin request policy ids in the cache behaviors block when creating a distribution in cloudfront_distribution.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
cloudfront_distribution

##### ADDITIONAL INFORMATION
Currently we are unable to set cache policy or origin request policy ids in the cache behaviors section so that they are added to a distribution.

Example playbook code:
```yaml
      cache_behaviors:
        - path_pattern: "*/"
          target_origin_id: ""
          field_level_encryption_id: ""
          cache_policy_id: "{{ cache_policy_id }}"
          origin_request_policy_id: "{{ origin_request_policy_id }}"

```

Example python code:
```python
cache_behavior['cache_policy_id'] = cache_behavior.get('cache_policy_id', config.get('cache_policy_id'))
cache_behavior['origin_request_policy_id'] = cache_behavior.get('origin_request_policy_id', config.get('origin_request_policy_id'))
```

### PR DESCRIPTION
**Pull request by Zozman**
##### SUMMARY
As described by issue #290, the current `cloudfront_distribution` module does not support `cache_policy_id` and `origin_request_policy_id` in a behavior.  In particular, attempting to use `cache_policy_id` will add default values that are incompatible with it (mainly `default_ttl`, `max_ttl`, `min_ttl`, and `compress`).  So this change allows you to use the policies instead of using a `forwarded_values` for a given behavior.

Fixed #290 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
cloudfront_distribution

### CODE REVIEW

**Comment by Zozman** [comment]
recheck

**Comment by Zozman** [comment]
recheck

**Comment by Zozman** [comment]
@markuman Thanks for the heads up; wouldn't surprise me if I messed it up so will try to see what's up.  Most likely I messed it up somewhere.

**Comment by Zozman** [comment]
@markuman @wwahammy Just did the rebase and seeing if now maybe this works or if I'm just dumb and did it wrong.

File: plugins/modules/cloudfront_distribution.py [hunk 1]
```diff
@@ -326,9 +340,23 @@
...
+            - A behavior should use either a C(cache_policy_id) or a C(forwarded_values) option.
+            - For more information see the CloudFront documentation
+              at U(https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-the-cache-key.html)
+          type: str
+        origin_request_policy_id:
+          description:
+            - The ID of the origin request policy for CloudFront to use for the cache behavior.
+            - For more information see the CloudFront documentation
+              at U(https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-origin-requests.html)
+          type: str
```
[the remark this answers is not in the document]
Comment (Zozman) [reply plugins/modules/cloudfront_distribution.py]: @markuman Done!

**Review by markuman — APPROVED** [review]

**Review by abikouo — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Comment by Zozman** [comment]
rebuild

**Comment by Zozman** [comment]
recheck

**Comment by Zozman** [comment]
ready_for_review

**Comment by Zozman** [comment]
Are the errors in the sanity tests something caused by my PR I can fix?  It looks like an issue with the job installing dependencies.

**Comment by markuman** [comment]
recheck

**Comment by wwahammy** [comment]
> Are the errors in the sanity tests something caused by my PR I can fix? It looks like an issue with the job installing dependencies.

Just an FYI, I am getting the same error when I install the branch locally using: `ansible-galaxy install -r roles/requirements.yml` with the requirements.yml of :

```yaml
---
# roles:
# Install a role from Ansible Galaxy.
# - name: sample
#   version: 1.9.6

collections:
  - name: https://github.com/zozman/community.aws.git # TODO this should be changed to upstream as described in https://github.com/walkerart/devops/issues/12
    type: git
    version: cloudfrontCachePolicy

```

**Comment by Zozman** [comment]
recheck

**Comment by Zozman** [comment]
recheck

**Comment by Zozman** [comment]
bot_status

**Pull request merged**